Core/UIViewExtension.swift

```diff
@@ -73,4 +73,17 @@ extension UIView {
             view.removeFromSuperview()
         }
     }
+    
+    @MainActor
+    public func createImageSnapshot(inBounds bounds: CGRect? = nil) -> UIImage? {
+        let bounds = bounds ?? self.frame
+        let size = bounds.size
+        UIGraphicsBeginImageContextWithOptions(size, false, UIScreen.main.scale)
+        UIGraphicsGetCurrentContext()?.translateBy(x: -bounds.origin.x, y: -bounds.origin.y)
+        drawHierarchy(in: frame, afterScreenUpdates: true)
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return image
+    }
```

Comment:
Any chance the [`snapshotView` API](https://developer.apple.com/documentation/uikit/uiview/1622531-snapshotview) can be used here instead?

Reply:
I tried that and wasn't getting anywhere with it (because the offset of the view matters and it wasn't clear how to use the snapshot API like that).  But I'll give it another shot as it'll be faster.

Reply:
Would require quite a bit of code to change tbh.  Would rather come back to this as an optimisation.

Reply:
Sounds good :+1: